src/App.tsx

```diff
@@ -15,36 +23,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const sortByLength = (arr: Goods, isReversed: boolean) => {
+  return arr.toSorted((a: string, b: string) => {
+    if (isReversed) {
+      return b.length - a.length;
+    }
+
+    return a.length - b.length;
+  });
+};
+
+const sortAlphabetically = (arr: Goods, isReversed: boolean) => {
+  if (isReversed) {
+    return arr.toSorted().reverse();
+  }
+
+  return arr.toSorted();
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Goods>(goodsFromServer);
+  const [sorting, setSorting] = useState<SortTypes | null>(null);
+  const [isReversed, setIsReveresed] = useState(false);
```

Comment:
There is a typo in the state variable name `setIsReveresed`. It should be `setIsReversed` to match the intended functionality and avoid confusion.